library/src/main/java/com/datatorrent/lib/io/block/ReaderContext.java

```diff
@@ -167,10 +174,54 @@ public LineReaderContext()
     @Override
     public void initialize(STREAM stream, BlockMetadata blockMetadata, boolean consecutiveBlock)
     {
+      overflowBlockRead = false;
+      posInStr = 0;
+      offset = blockMetadata.getOffset();
+      super.initialize(stream, blockMetadata, consecutiveBlock);
+    }
+
+    /**
+     * Reads bytes from the stream starting from the offset into the buffer
+     *
+     * @param bytesFromCurrentOffset
+     *          bytes read till now from current block
+     * @param bytesToFetch
+     *          the number of bytes to be read from stream
+     * @return the number of bytes actually read, -1 if 0 bytes read
+     * @throws IOException
+     */
+    protected int readData(final long bytesFromCurrentOffset, final int bytesToFetch) throws IOException
+    {
       if (buffer == null) {
-        buffer = new byte[bufferSize];
+        buffer = new byte[bytesToFetch];
       }
-      super.initialize(stream, blockMetadata, consecutiveBlock);
+      return stream.read(offset + bytesFromCurrentOffset, buffer, 0, bytesToFetch);
+    }
+
+    /**
+     * @param usedBytesFromOffset
+     *          number of bytes the pointer is ahead of the offset
+     * @return true if end of stream reached, false otherwise
+     */
+    protected boolean checkEndOfStream(final long usedBytesFromOffset)
+    {
+      if (!overflowBlockRead) {
+        return (offset - blockMetadata.getOffset() + usedBytesFromOffset < bufferSize);
+      } else {
```

Comment:
I think offset and blockMetadata.getOffset() are equal. If yes, then you can reduce the expression. Please correct it, if i am wrong.